mbozee/session_3/mailroom.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+
+donors = {
+    'Donovan Leitch': [100, 20, 1000],
+    'Stevie Nicks': [200, 500],
+    'Ian Anderson': [190],
+    'Mavis Staples': [170, 40, 130],
+    'Suzanne Vega': [160, 300]
+}
+
+
+def menu():
+    """Give user choice of actions."""
+    for keys, values in donors.items():
+        print(keys, end=', ')
+    print(end='\n')
+    options = {1: 'Send a Thank You', 2: 'Create a Report', 3: 'quit'}
+    choice = input("Choose an option:\n" + str(options) + " > ")
+
+    while choice != '3' and choice != 'quit'.lower():
+
+        if choice == str(1):
+            thank_you()
+        elif choice == str(2):
+            create_report()
+        else:
+            menu()
+        break
+
+
+def thank_you():
+    """Write a thank you letter."""
+    full_name = input("(Donor full name) / (list) > ")
+
+    if full_name == 'list'.lower():
+        print("Donors:", end=' ')
+        for key, value in donors.items():
+            print(key, end=', ')
+        print('\n')
+        thank_you()
+    elif full_name in donors:
+        new_donation = int(input("New donation amount > "))
```

Comment:
Currently if i enter string that can't be cast to an int, and exception is thrown.  The above code will catch that and re-prompt.